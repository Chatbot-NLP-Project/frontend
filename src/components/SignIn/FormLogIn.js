import React, { useEffect } from "react";
import validate from "./validateInfo";
import useForm from "./useForm";
import "./LogIn.css";
// Promise based HTTP client for the node.js
//import Axios from "axios";
import { Badge, Box, Button, Center } from "@chakra-ui/react";

function FormLogIn(props) {
  const { handleChange, handleSubmit, values, errors, backEndErrors } = useForm(
    props.submitForm,
    validate,
    props.setSignInClicked
  );

  return (
    <div className="login-content-right">
      <form onSubmit={handleSubmit} className="form" noValidate>
        <h1>
          Welcome!!!
          <br />
          Sign in to use our service!
          <br />
        </h1>
        {backEndErrors == "" ? (
          ""
        ) : (
          <Box w="80%" h="40px" bg="red.500" borderRadius="5px">
            <Center>
              <Badge variant="solid" m="2%" colorScheme="red">
                {backEndErrors}
              </Badge>
            </Center>
          </Box>
        )}
        <div className="login-inputs">
          <label className="login-label">Email</label>
          <input
            className="login-input"
            type="email"
            name="email"
            placeholder="Enter your email"
            value={values.email}
            onChange={handleChange}
          />
          {errors.email && <p>{errors.email}</p>}
        </div>
        <div className="login-inputs">
          <label className="login-label">Password</label>
          <input
            className="login-input"
            type="password"
            name="password"
            placeholder="Enter your password"
            value={values.password}
            onChange={handleChange}
          />
          {errors.password && <p>{errors.password}</p>}
        </div>
        {backEndErrors == "" ? (
          ""
        ) : (
          <Box h='10%' mt='20px'>
            <Badge variant="outline" colorScheme="red">
              Sign In Failed
            </Badge>
          </Box>
        )}
        <button className="login-input-btn" type="submit">
          Sign In
        </button>
        <span className="login-input-login">
          Don't have an account? Sign up <a href="signup">here</a>
        </span>
      </form>
    </div>
  );
}

export default FormLogIn;
