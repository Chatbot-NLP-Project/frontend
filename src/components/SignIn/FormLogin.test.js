import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import FormLogIn from "./FormLogIn"
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

test('on initial render, form is there', () => {
    render(<FormLogIn />) //submitForm = {false} setSignInClicked={false}
    screen.debug()
})

test('buttons are enabled', () => {
    render(<FormLogIn />) //submitForm = {false} setSignInClicked={false}
    expect(screen.getByRole('button', {name: /continue/i})).toBeEnabled();
})

test('add inputs to email and password', () => {
    render(<FormLogIn />) //submitForm = {false} setSignInClicked={false}
    
    userEvent.type(screen.getByPlaceholderText(/Enter your email/i), "chalindu@gmail.com");
    userEvent.type(screen.getByPlaceholderText(/Enter your password/i), "0715956909") ;
    userEvent.click(screen.getByRole('button', {name: /continue/i}));
})

test("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><FormLogIn /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})