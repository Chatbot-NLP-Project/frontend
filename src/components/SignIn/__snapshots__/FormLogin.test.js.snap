// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="login-content-right"
>
  <form
    className="form-signin"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h1
      className="form-signin-h1"
      color="black"
    >
      <br />
      Sign in to use our service!
      <br />
    </h1>
    
    <div
      className="login-inputs"
    >
      <label
        className="login-label"
      >
        Email
      </label>
      <input
        className="login-input"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      className="login-inputs"
    >
      <label
        className="login-label"
      >
        Password
      </label>
      <input
        className="login-input"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    
    <button
      className="login-input-btn"
      type="submit"
    >
      Continue
    </button>
    <span
      className="login-input-login"
    >
      Don't have an account? Sign up 
      <a
        href="signup"
      >
        here
      </a>
    </span>
  </form>
</div>
`;
